### ISSUE TITLE
Remark42 doesn't work behind Apache2 reverse proxy when in subdirectory

### ISSUE THREAD

**Issue by mind-overflow** [umputun/remark42#759]
Hello, and thanks for being such awesome devs!
Unfortunately, I'm having problems setting up Remark42 behind an Apache2 reverse proxy.
I read this wiki page: [[How to configure remark42 with nginx reverse proxy](https://github.com/umputun/remark42/wiki/How-to-configure-remark42-with-nginx-reverse-proxy)], and proceeded translating it to Apache.
After setting remark42 to port 8089, this is the very minimal setup I ended up with:

```
<VirtualHost *:80>
        ServerName comments.domain.net
        RewriteEngine on
        RewriteCond %{SERVER_NAME} = comments.domain.net
        RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
</VirtualHost>

<VirtualHost *:443>
        ServerName comments.domain.net

        ProxyRequests Off
        ProxyPreserveHost On
        ProxyVia Full
        <Proxy *>
                Require all granted
        </Proxy>

        RequestHeader set X-Forwarded-Proto "https"
        RequestHeader set X-Forwarded-Port "443"

        ProxyPass "/" "http://127.0.0.1:8089/"
        ProxyPassReverse "/" "http://127.0.0.1:8089/"

        Include *******
        SSLCertificateFile *******
        SSLCertificateKeyFile *******
</VirtualHost>
```

Which works flawlessly in a subdomain.
However, my main goal was to set it up under a subdirectory, such as `services.domain.net/comments/`.
This is the wiki page explaining how to do it with nginx: [[How to configure remark42 without a subdomain](https://github.com/umputun/remark42/wiki/How-to-configure-remark42-without-a-subdomain)].
Following the guide, I set Remark42's URL to `https://services.domain.net/comments/`, and modified the Apache2 reverse proxy as follows:
```
<VirtualHost *:80>
        ServerName services.domain.net
        RewriteEngine on
        RewriteCond %{SERVER_NAME} = services.domain.net
        RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
</VirtualHost>

<VirtualHost *:443>
        ServerName services.domain.net

        ProxyRequests Off
        ProxyPreserveHost On
        ProxyVia Full
        <Proxy *>
                Require all granted
        </Proxy>

        <Location "/comments">
                RequestHeader set X-Forwarded-Proto "https"
                RequestHeader set X-Forwarded-Port "443"

                ProxyPass "http://127.0.0.1:8089/"
                ProxyPassReverse "http://127.0.0.1:8089/"
        </Location>

        Include *******
        SSLCertificateFile *******
        SSLCertificateKeyFile *******
</VirtualHost>
```

However, this did not work.
I tried different things such as adding or removing trailing slashes, using
```
                ProxyPass "http://127.0.0.1:8089/comments/"
                ProxyPassReverse "http://127.0.0.1:8089/comments/"
```
instead of
```
                ProxyPass "http://127.0.0.1:8089/"
                ProxyPassReverse "http://127.0.0.1:8089/"
```

etc. However, visiting `https://services.domain.net/comments/web/` always resulted in a 404 error.

P.S.: I am running remark42 with docker-compose, and I have exposed port 8089.
Thanks!

**Comment by umputun** [comment]
I have not touched apache in years, but my guess would be - your location is not matching the url. It has `Location "/services"` but you access `/comments`. I think you should change it to `<Location "/comments">`

**Comment by mind-overflow** [comment]
> I have not touched apache in years, but my guess would be - your location is not matching the url. It has `Location "/services"` but you access `/comments`. I think you should change it to `<Location "/comments">`

Whoops... Actually, that was a mistake I did by copying the config to GitHub. Since I couldn't get remark to work under the subdir, I have temporarily switched back to the subdomain. So I recreated the config here, and accidentally messed up the location tag.
The actual live config had `<Location "/comments">`. However, I just tried again and it still returns 404.

**Comment by umputun** [comment]
I just created a similar, working configuration but put apache to the container as well as I don't have it installed on any host. The only important thing I noticed - in order to make all urls with "/comments" prefix work, I had to change location to`<Location "/comments/">` (added `/` suffix)

This is what my proxy location looks like:

```
        <Location "/comments/">
                ProxyPass "http://remark42:8080/"
                ProxyPassReverse "http://remark42:8080/"
        </Location>
```

Note: I have here `remark42` as a host because both apache and remark42 run inside of the same compose's namespace. For your external apache it should be `http://127.0.0.1:8089/`

If still doesn't work I'd suggest some more troubleshooting steps:

- try to hit remark42 /ping, i.e. https://services.domain.net/comments/ping and see if it works. If it is - you reaching remark42
- try to hit remark42 web directly, i.e. "http://127.0.0.1:8089/web/`

_the bottom line: I don't think the issue you described has anything to do with remark42, but rather apache reverse proxy configuration_

**Comment by mind-overflow** [comment]
The problem was the trailing slash in the Location tag - `<Location "/comments/">`.
After adding it, everything started working flawlessly.
Thank you so much for the help, and sorry for bothering you. I really couldn't get my head around it.

Maybe, as a little side note, now that we know it works, you could add those two configurations to the Wiki!
Thanks again.
